java-frontend/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -87,6 +89,8 @@ public String valueAsString() {
   public String excludedTypes = "";
   private final List<String> excludedTypesList = new ArrayList<>();
 
+  private final Set<TryStatementTree> tryWithResourcesTrees = new HashSet<>();
```

Comment:
It was not clear to me that the content of this set was used only to know if you already collected the resources from the try. Maybe renaming it as `visitedTryWithResourcesTrees ` or something like this would help?